src/tox_current_env/hooks4.py

```diff
@@ -0,0 +1,241 @@
+
+    def install(self, *args, **kwargs):
+        return None
+
+
+class CurrentEnvLocalSubProcessExecutor(Execute):
+    def build_instance(
+        self,
+        request,
+        options,
+        out,
+        err,
+    ):
+        request.env["PATH"] = ":".join(
+            (str(options._env.env_dir / "bin"), request.env.get("PATH", ""))
+        )
+        return LocalSubProcessExecuteInstance(request, options, out, err)
+
+
+class CurrentEnv(PythonRun):
+    def __init__(self, create_args):
+        self._executor = None
+        self._installer = None
+        self._path = []
+        super().__init__(create_args)
+
+    @staticmethod
+    def id():
+        return "current-env"
+
+    @property
+    def _default_package_tox_env_type(self):
+        return None
+
+    @property
+    def _external_pkg_tox_env_type(self):
+        return None
+
+    @property
+    def _package_tox_env_type(self):
+        return None
+
+    @property
+    def executor(self):
+        if self._executor is None:
+            self._executor = CurrentEnvLocalSubProcessExecutor(self.options.is_colored)
+        return self._executor
+
+    def _get_python(self, base_python):
+        return PythonInfo(
+            implementation=sys.implementation,
+            version_info=sys.version_info,
+            version=sys.version,
+            is_64=(platform.architecture()[0] == "64bit"),
+            platform=platform.platform(),
+            extra={"executable": Path(sys.executable)},
+        )
+
+    def create_python_env(self):
+        """Fake Python environment just to make sure all possible
+        commands like python or python3 works."""
+        bindir = self.env_dir / "bin"
+        if not bindir.exists():
+            os.mkdir(bindir)
+        for suffix in (
+            "",
+            f"{sys.version_info.major}",
+            f"{sys.version_info.major}.{sys.version_info.minor}",
+        ):
+            symlink = bindir / f"python{suffix}"
+            if not symlink.exists():
+                os.symlink(sys.executable, symlink)
+
+    def env_bin_dir(self):
+        return Path(sysconfig.get_path("scripts"))
+
+    def env_python(self):
+        return sys.executable
+
+    def env_site_package_dir(self):
+        return Path(sysconfig.get_path("purelib"))
+
+    @property
+    def installer(self):
+        return Installer()
+
+    def prepend_env_var_path(self):
+        return [self.env_bin_dir()]
+
+    @property
+    def runs_on_platform(self):
+        return sys.platform
+
+
+class PrintEnv(CurrentEnv):
+    def __init__(self, create_args):
+        super().__init__(create_args)
+
+        if self.options.print_extras_to:
+            if "extras" not in self.conf:
+                # Unfortunately, if there is skipsdist/no_package or skip_install
+                # in the config, this section is not parsed at all so we have to
+                # do it here manually to be able to read its content.
+                self.conf.add_config(
```

Comment:
Why do you need this if skipsdist/no_package?

Reply:
Because we need to get the information if specified. I guess defining extras together with skipsdist/no_package is invalid -- does it raise an error?

Reply:
> Because we need to get the information if specified. I g

Why?

> I guess defining extras together with skipsdist/no_package is invalid -- does it raise an error?

It's not invalid, but there's no package involved in these cases, so there's no target to find extras for?

Reply:
Becasu the purpose of the `--print-extras-to` option is to get the list of extras so we can gather the requirements for tests and install them via dnf in rpm build environment.

Reply:
> Becasu the purpose of the `--print-extras-to` option is to get the list of extras so we can gather the requirements for tests and install them via dnf in rpm build environment.

It sounds like you want to execute a `tox deps -e py310` that would print the dependencies, taking into consideration the extras specified for the env.

Reply:
`tox deps` does not seem to exist in tox 4.0.9 `tox depends` outputs something I don't understand at all.

Reply:
tox deps would be a new feature not yet existing today 😆

Reply:
oh, I misunderstood that

Reply:
If deps is to be implemented we could use it only if it was possible to determine all the dependencies without actually building a wheel (or using the appropriate metadata hook) from the project. If it were to include the deps of extras, I am not sure that is technically possible. For our purposes it is much easier to get the list of direct dependencies and a list of extras and use the list of extras to query the actual package we are building (somewhere else) for the runtime dependencies including those extras.

Reply:
Pep-621 projects can be determined without building a wheel, but otherwise you'd need to build wheel or use the metadata hook to determine extras, no?

Reply:
That is exactly why we just want the list of extras from tox. We are building the wheel/metadata elsewhere differently and will use the list of extras extracted from tox. I.e. we don't want tox to build the wheel/metadata and provide the specific dependencies, we just need the list of extras specified in the config.

Reply:
For that, you could use:

```
tox c -k extras
```

and parse it within an INI parser?

Reply:
That way, you would not miss changes to the extra parse we're adding, e.g. https://github.com/tox-dev/tox/pull/2668; and is the more supported way of reading tox configuration.

Reply:
Yes, that seems possible, unless the tox configuration changes. Whether or not the extras are normalized does not matter to us, we normalize them anyway.